tox.ini

```diff
@@ -22,7 +22,40 @@ passenv =
   CHARM_BUILD_DIR
   MODEL_SETTINGS
 
+[testenv:fmt]
+description = Apply coding style standards to code
+deps =
+    black
+    isort
+commands =
+    isort {[vars]all_path}
+    black {[vars]all_path}
+
+[testenv:lint]
+description = Check code against coding style standards
+deps =
+    black
+    flake8
+    flake8-docstrings
+    flake8-copyright
+    flake8-builtins
+    pyproject-flake8
+    pep8-naming
+    isort
+    codespell
+commands =
+    # uncomment the following line if this charm owns a lib
+    # codespell {[vars]lib_path}
+    codespell {toxinidir}/. --skip {toxinidir}/.git --skip {toxinidir}/.tox \
+      --skip {toxinidir}/build --skip {toxinidir}/lib --skip {toxinidir}/venv \
+      --skip {toxinidir}/.mypy_cache --skip {toxinidir}/icon.svg
+    # pflake8 wrapper supports config from pyproject.toml
+    pflake8 {[vars]all_path}
+    isort --check-only --diff {[vars]all_path}
+    black --check --diff {[vars]all_path}
```

Comment:
Should we also add pylint and mypy?

Reply:
The tox configuration is shared accross charms, so it'd better be done here https://github.com/canonical/charmcraft/blob/main/charmcraft/templates/init-simple/tox.ini.j2

Reply:
We can leave this for another PR, we have already deviated from the `tox.ini` file by adding black for a good reason that we needed to have additional security checks. A similar argument could apply to `pylint`, `mypy` and potentially `pydocstyle` if we believe there is value in the additional checks

Reply:
Indeed, but we should probably look into converging again by contributing there

Reply:
Let's continue this discussion outside the PR